SemiLeptonicWVA/Utils/src/GenEventInfo.cc

```diff
@@ -264,6 +270,23 @@ GenEventInfo::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 	  PUWeight = LumiWeights_.weight( Tnpv );
 	}
 
+
+
+	edm::Handle< LHEEventProduct > LHEEventInfo;
+        iEvent.getByLabel("externalLHEProducer", LHEEventInfo);
+
+	std::vector<float> genweights;
```

Comment:
Change all the 'float' to 'double' types, just to be safe (we are talking about values of e-08).  Also, change 'genweights' to 'AQGCweights' to be more specific, because we already have a similar variable in the event that expresses if the event has positive or negative weight.